### ISSUE TITLE
Enchanting wont work in survival mode (1.20.1)

### ISSUE THREAD

**Issue by Lumguava** [EndLone/MCDE#37]
Hi! I'd like to report a bug regarding rerolls. There's a weird behavior when trying to reroll enchants in survival and it **won't recognize lapis lazuli but echo shards only**. Although it accepts the echo shard, the actual **enchantment and slots won't update when rerolling** unless you go to creative mode.

I tried to find for weird logs in my server but it does not produce any when I'm rerolling. Can you help?

https://github.com/user-attachments/assets/78e86b86-bdc7-4388-a176-c808d01de3b1

**Comment by Lumguava** [comment]
Nevermind, I think I got a better understanding of how the mod works.

Lapis for swapping enchantments and Echo Shards for slots. I thought it was bugged due to low chances of more slots with echo shards. Figured out the story advancements had to do something with increasing chances. My fault!

I've already tweaked the config just now to give more chance of getting slots. Thank you regardless! This mod is really phenomenal.

**Comment by Lumguava** [comment]
Nevermind, I think I got a better understanding of how the mod works.

Lapis for swapping enchantments and Echo Shards for slots. I thought it was bugged due to low chances of more slots with echo shards. Figured out the story advancements had to do something with increasing chances. My fault!

I've already tweaked the config just now to give more chance of getting slots. Thank you regardless! This mod is really phenomenal.
